src/jit/morph.cpp

```diff
@@ -10831,10 +10829,7 @@ GenTreePtr          Compiler::fgMorphSmpOp(GenTreePtr tree, MorphAddrContext* ma
 
                 if (andOp->gtType == TYP_INT)
                 {
-                    if (!andOp->gtOp.gtOp2->IsCnsIntOrI())
-                        goto SKIP;
-
-                    if (andOp->gtOp.gtOp2->gtIntCon.gtIconVal != 1)
+                    if (!andOp->IsIOne())
```

Comment:
This change introduces a correctness bug as the previous condition was on andop->gtOp.gtOp2 (i.e. second operand of andOp whereas this is on andOp itself.